exchanges/huobi/huobi_websocket.go

```diff
@@ -75,427 +73,240 @@ var subscriptionNames = map[string]string{
-	case strings.Contains(init.Channel, "depth"):
-		var depth WsDepth
-		err := json.Unmarshal(respRaw, &depth)
-		if err != nil {
-			return err
-		}
+	h.Websocket.DataHandler <- stream.UnhandledMessageWarning{
+		Message: h.Name + stream.UnhandledMessage + string(respRaw),
+	}
 
-		data := strings.Split(depth.Channel, ".")
-		err = h.WsProcessOrderbook(&depth, data[1])
-		if err != nil {
-			return err
-		}
-	case strings.Contains(init.Channel, "kline"):
-		var kline WsKline
-		err := json.Unmarshal(respRaw, &kline)
-		if err != nil {
-			return err
-		}
-		data := strings.Split(kline.Channel, ".")
-		var p currency.Pair
-		var a asset.Item
-		p, a, err = h.GetRequestFormattedPairAndAssetType(data[1])
-		if err != nil {
-			return err
-		}
-		h.Websocket.DataHandler <- stream.KlineData{
-			Timestamp:  time.UnixMilli(kline.Timestamp),
-			Exchange:   h.Name,
-			AssetType:  a,
-			Pair:       p,
-			OpenPrice:  kline.Tick.Open,
-			ClosePrice: kline.Tick.Close,
-			HighPrice:  kline.Tick.High,
-			LowPrice:   kline.Tick.Low,
-			Volume:     kline.Tick.Volume,
-			Interval:   data[3],
-		}
-	case strings.Contains(init.Channel, "trade.detail"):
-		if !h.IsSaveTradeDataEnabled() {
-			return nil
-		}
-		var t WsTrade
-		err := json.Unmarshal(respRaw, &t)
-		if err != nil {
-			return err
-		}
-		data := strings.Split(t.Channel, ".")
-		var p currency.Pair
-		var a asset.Item
-		p, a, err = h.GetRequestFormattedPairAndAssetType(data[1])
-		if err != nil {
-			return err
-		}
-		var trades []trade.Data
-		for i := range t.Tick.Data {
-			side := order.Buy
-			if t.Tick.Data[i].Direction != "buy" {
-				side = order.Sell
-			}
-			trades = append(trades, trade.Data{
-				Exchange:     h.Name,
-				AssetType:    a,
-				CurrencyPair: p,
-				Timestamp:    time.UnixMilli(t.Tick.Data[i].Timestamp),
-				Amount:       t.Tick.Data[i].Amount,
-				Price:        t.Tick.Data[i].Price,
-				Side:         side,
-				TID:          strconv.FormatFloat(t.Tick.Data[i].TradeID, 'f', -1, 64),
-			})
-		}
-		return trade.AddTradesToBuffer(h.Name, trades...)
-	case strings.Contains(init.Channel, "detail"),
-		strings.Contains(init.Rep, "detail"):
-		var wsTicker WsTick
-		err := json.Unmarshal(respRaw, &wsTicker)
-		if err != nil {
-			return err
-		}
-		var data []string
-		if wsTicker.Channel != "" {
-			data = strings.Split(wsTicker.Channel, ".")
-		}
-		if wsTicker.Rep != "" {
-			data = strings.Split(wsTicker.Rep, ".")
-		}
+	return nil
+}
 
-		var p currency.Pair
-		var a asset.Item
-		p, a, err = h.GetRequestFormattedPairAndAssetType(data[1])
-		if err != nil {
-			return err
-		}
+func (h *HUOBI) wsHandleV2ping(_ string, respRaw []byte) error {
+	ts, err := jsonparser.GetInt(respRaw, "data", "ts")
+	if err != nil {
+		return fmt.Errorf("error getting ts from auth ping: %w", err)
+	}
+	if err := h.Websocket.AuthConn.SendJSONMessage(context.Background(), request.Unset, json.RawMessage(`{"action":"pong","data":{"ts":`+strconv.Itoa(int(ts))+`}}`)); err != nil {
```

Comment:
This is auth only? Maybe inject the connection into the param if not and upgrade wsHandleData params.

Reply:
Yes. It's auth only.
Note the error messages if unsure.

Refactored for clarity.

Fixed https://github.com/gbjk/gocryptotrader/commit/a89f6f8f0